insights/client/connection.py

```diff
@@ -175,6 +201,57 @@ def _init_session(self):
                 connection.proxy_headers = auth_map
         return session
 
+    def _http_request(self, url, method, **kwargs):
+        '''
+        Perform an HTTP request, net logging, and error handling
+
+        Parameters
+            url     - URL to perform the request against
+            method  - HTTP method, used for logging
+            kwargs  - Rest of the args to pass to the request function
+        Returns
+            HTTP response object on completion
+        Side effects
+            Calls handle_fail_rcs to handle messaging and exceptions for failure codes
+            Raises RuntimeError on timeout or failure to connect
+        '''
+        try:
+            logger.log(NETWORK, "%s %s", method, url)
+            res = self.session.request(url=url, method=method, timeout=self.config.http_timeout, **kwargs)
+            logger.log(NETWORK, "HTTP Status Code: %d", res.status_code)
+            logger.log(NETWORK, "HTTP Status Text: %s", res.reason)
+            logger.log(NETWORK, "HTTP Response Text: %s", res.text)
+            res.raise_for_status()
+            return res
+        except requests.Timeout as e:
+            logger.error(e)
+            raise TimeoutException("Connection timed out.")
+        except requests.ConnectionError as e:
+            logger.error(e)
+            raise RuntimeError("The Insights API could not be reached.")
+        except requests.exceptions.HTTPError as e:
+            logger.error(e)
+            self.handle_fail_rcs(res)
```

Comment:
This could be just as well outside the _except_ block along with the _return_. Despite its name, the function more or less handles the response by its status code, which does not necessarily be a failure one. Well, it’s ok like this too.